datagovsg/send
# feat: add check middleware to optionally block non-OGP users

## Description
In order to prevent intranet users from using Send staging to cross the internet/intranet divide. We want to block them on staging.

To do this I've opted to add a check to the session middleware dependent on an env flag. If enabled, it will check redirect non-OGP users back to main Vault, preventing them from using Send.

## How to test
- checkout
- copy `export BLOCK_NON_OGP="true"` from env_example to local.env
- login as an @open.gov.sg email, navigate to send, it should work as normal
- login as an @tech.gov.sg email, navigate to send, it should redirect to the main Vault landing page
